bundles/org.openhab.core.config.discovery.addon.usb/src/main/java/org/openhab/core/config/discovery/addon/usb/UsbAddonFinder.java

```diff
@@ -0,0 +1,186 @@
+
+    private final Logger logger = LoggerFactory.getLogger(UsbAddonFinder.class);
+    private final Set<UsbSerialDiscovery> usbSerialDiscoveries = new CopyOnWriteArraySet<>();
+    private final Map<Long, UsbSerialDeviceInformation> usbDeviceInformations = new ConcurrentHashMap<>();
+
+    @Reference(cardinality = ReferenceCardinality.MULTIPLE, policy = ReferencePolicy.DYNAMIC)
+    protected void addUsbSerialDiscovery(UsbSerialDiscovery usbSerialDiscovery) {
+        usbSerialDiscoveries.add(usbSerialDiscovery);
+        usbSerialDiscovery.registerDiscoveryListener(this);
+        usbSerialDiscovery.doSingleScan();
+    }
+
+    protected synchronized void removeUsbSerialDiscovery(UsbSerialDiscovery usbSerialDiscovery) {
+        usbSerialDiscovery.unregisterDiscoveryListener(this);
+        usbSerialDiscoveries.remove(usbSerialDiscovery);
+    }
+
+    @Override
+    public Set<AddonInfo> getSuggestedAddons() {
+        Set<AddonInfo> result = new HashSet<>();
+        for (AddonInfo candidate : addonCandidates) {
+            for (AddonDiscoveryMethod method : candidate.getDiscoveryMethods().stream()
+                    .filter(method -> SERVICE_TYPE.equals(method.getServiceType())).toList()) {
+                Map<String, Pattern> matchProperties = method.getMatchProperties().stream()
+                        .collect(Collectors.toMap(property -> property.getName(), property -> property.getPattern()));
+
+                Set<String> propertyNames = new HashSet<>(matchProperties.keySet());
+                propertyNames.removeAll(SUPPORTED_PROPERTIES);
+
+                if (!propertyNames.isEmpty()) {
+                    logger.warn("Add-on '{}' addon.xml file contains unsupported 'match-property' [{}]",
+                            candidate.getUID(), String.join(",", propertyNames));
+                    break;
+                }
+
+                logger.trace("Checking candidate: {}", candidate.getUID());
+                for (UsbSerialDeviceInformation device : usbDeviceInformations.values()) {
+                    logger.trace("Checking device: {}", device);
+
+                    if (propertyMatches(matchProperties, PRODUCT, device.getProduct())
+                            && propertyMatches(matchProperties, MANUFACTURER, device.getManufacturer())
+                            && propertyMatches(matchProperties, CHIP_ID,
+                                    toHexString(device.getVendorId(), device.getProductId()))
+                            && propertyMatches(matchProperties, REMOTE, String.valueOf(device.getRemote()))) {
+                        result.add(candidate);
+                        logger.debug("Suggested add-on found: {}", candidate.getUID());
+                        break;
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    private String toHexString(int vendorId, int productId) {
+        return String.format("0x%04x:0x%04x", vendorId, productId);
```

Comment:
It's more common to see these string in output of tools like lsusb without these `0x` prefixes.

```suggestion
        return String.format("%04x:%04x", vendorId, productId);
```

Reply:
Ok